docs/advanced-features/i18n-routing.md

```diff
@@ -204,6 +204,30 @@ export default function IndexPage(props) {
 }
 ```
 
+Note that when using [dynamic routes](/docs/routing/dynamic-routes) you will need to pass the `href` argument of `router.push` as an object.
+```jsx
+import { useRouter } from 'next/router'
+
+export default function IndexPage(props) {
+  const router = useRouter()
+
+  return (
+    <div
+      onClick={() => {
+        router.push({
+          pathname: '/blog/[post]',
```

Comment:
```suggestion
          pathname: '/post/[post]',
```
It looks like this dynamic route is mis-matching the `as` value, let's make sure this matches to prevent confusion.